js/apps/system/_admin/aardvark/APP/react/src/components/form/CreatableMultiSelectControl.tsx

```diff
@@ -17,12 +17,15 @@ export const CreatableMultiSelectControl = (props: InputControlProps) => {
   const [field, , helper] = useField(name);
   const { isSubmitting } = useFormikContext();
 
-  let value = selectProps?.options?.filter(option => {
-    return (option as OptionType).value === field.value;
-  }) as PropsValue<OptionType>;
+  let value =
```

Comment:
I think this logic can be simplified, what do you think about this alternative?

```
let value = [];

if (typeof field.value === "string") {
  value = (selectProps?.options || []).filter(option => option.value === field.value);
} else if (!selectProps?.options && field.value) {
  value = field.value.map((value) => ({ value, label: value }));
}
```

Reply:
moved this whole logic into a separate function, it was getting too convoluted